keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -111,6 +112,35 @@ def __init__(self, img_height=512, img_width=512, jit_compile=False):
         self.diffusion_model.load_weights(diffusion_model_weights_fpath)
         self.decoder.load_weights(decoder_weights_fpath)
 
+    def add_tokens(self, tokens):
```

Comment:
This should not be on this class IMO. Time to refactor `StableDiffusion` to enable a growing set of modular use cases.

`StableDiffusion` should hold the 3 base models + a public interface. Then the actual logic should be in classes such as `TextToImage`, etc (or perhap in plain functions living somewhere like `text_to_image.py`). Something like

```python
# in class StableDiffusion
def text_to_image(self, *args, **kwargs):
   return text_to_image(self.encoder, self.decoder, self.diffusion_model, *args, **kwargs)
```

This enables growing our set of uses cases while keeping the logic modular and manageable.

Reply:
Sounds good.  I'll refactor in this pr.

Reply:
While writing the tutorial and doing research on this I found that people tend to fetch an initialization for another similar object.

so for <my-silly-cat-token> you’d initialize with the embedding for “cat”, and for “Luke” you’d initialize with “human”

Reply:
That makes sense, you want an initialization mechanism that minimizes the distance the embedding will have to travel during fine-tuning.

Reply:
So, would you suggest along with the refactor mentioned about that this method live outside of stable diffusion (perhaps in the keras.io tutorial?).  If so, then we'd have to export the sub-models from KerasCV, which might be a little ugly.

Reply:
Currently I'm thinking:

keep add_tokens() in the tokenizer, move the rest to the guide.  This will let us actually show off the weight initialization scheme via adjacent token, etc.